assets/src/apps/delivery/store/features/page/actions/loadInitialPageState.ts

```diff
@@ -131,9 +130,8 @@ export const loadInitialPageState = createAsyncThunk(
 
       const shouldResume = attempts.some((attempt: any) => attempt.dateEvaluated !== null);
       if (shouldResume && !isReviewMode) {
-        // state should be all up to date by now
-        const snapshot = getEnvState(defaultGlobalEnv);
-        const resumeId = snapshot['session.resume'];
+        //sessionState variable already have this info so lets get it from there because sometimes the defaultGlobalEnv state was not up to date by now
```

Comment:
`const resumeId = snapshot['session.resume']; ` was coming `'undefined'` and hence as the next code, the `resumeId ` was getting set to the next sequence in the list instead of the actual resume activity which was the reason behind resuming in wrong lesson.